framework/src/media/SocketOutputDataSource.cpp

```diff
@@ -100,7 +100,7 @@ bool SocketOutputDataSource::isPrepare()
 ssize_t SocketOutputDataSource::write(unsigned char* buf, size_t size)
 {
 	if (!buf) {
-		return (size_t)0;
+		return (ssize_t)0;
```

Comment:
Is typecasting needed in this case?

Reply:
typedef uint16_t size_t;
typedef int16_t ssize_t;

It looks like to execute a downcasting.
Isn't there any possibility about data missing?

Reply:
i think

"return EOF;"
This seems to be right.

Reply:
I think so "return EOF" and typecasting does not seem necessary.

How about FileOutputDataSource::write and BufferOutputDataSource::write?
FileOutputDataSource::write returns `(ssize_t)EOF`
BufferOutputDataSource::write returns `EOF`

Reply:
I think type casting is not neccessary, EOF is -1.
Also those of 3 datasources should return EOF.
(Of course we have to define some error types in future)

Reply:
size_t is just a parameter's type. Return value doesn't need to be same type with a parameter's type. About data loss, The function logic has to control it. This is not a type casting problem I think.